### PR TITLE
Term change WAF to ContentFilter

### PR DESCRIPTION
**Pull request by Ronyk1**
According to issue [#552](https://github.com/curiefense/curiefense/issues/552) the terms WAF Policies and WAF Rules were changed to Content Filter Profiles and Content Filter Rules.

### Changelog
The term was changed for:
- UI
- URL Paths
- CurieConf and CurieProxy variables and functions names
- Filenames
- Kibana and Grafana Dashboards

Also, added to confserver util functions to handle with name changing of fields in models.
### TODO
Update docs

### CODE REVIEW

**Review by bartavelle — APPROVED** [review]
Rust part LGTM

**Review by flaper87 — COMMENTED** [review]
Overall, I think this is fine. 

I think this will break backwards compatibility, tho. @Freakachoo @bartavelle @Aviv-Galmidi I would appreciate if you could take a few mins to check this too. Special attention in the integration between the Rust bits and this change.

**Comment by bartavelle** [comment]
The rust changes look OK to me, as they are pretty mechanical. The main issue would indeed be not to forget any bit ...

File: .github/workflows/e2e-minikube-filebeat.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
...
...test_ratelimit_event_query or test_ratelimit_event_method or test_ratelimit_event_authority or test_ipv6 or test_country or test_asn or test_ratelimit_scope_include or test_ratelimit_scope_include_exclude or test_ratelimit_scope_exclude or test_ratelimit_countby_section or test_ratelimit_countby2_section or test_ratelimit_countby_2sections or test_ratelimit_event_section or test_length_overlong or test_length_short or test_count_few or test_count_toomany or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_allowlisted_value or test_non_allowlisted_value_restrict or test_non_allowlisted_value_norestrict_no_content_filter_match or test_non_allowlisted_value_norestrict_content_filter_match or test_non_allowlisted_value_norestrict_content_filter_match_excludesig or test_allowlisted_value or test_allowlisted_value or test_non_allowlisted_value_restrict or test_non_allowlisted_value_norestrict_no_content_filter_match or test_non_allowlisted_value_norestrict_content_filter_match or test_non_allowlisted_value_norestrict_content_filter_match_excludesig)' .
```
[the remark this answers is not in the document]
Comment (Ronyk1) [reply .github/workflows/e2e-minikube-filebeat.yml:62]: No new tests excluded, only names were changed

File: curiefense/curieconf/server/curieconf/confserver/v1/api.py [hunk 2]
```diff
@@ -334,27 +339,27 @@ def validateJson(json_data, schema_type):
 urlmaps_file_path = (base_path / "./json/url-maps.schema").resolve()
 with open(urlmaps_file_path) as json_file:
     urlmaps_schema = json.load(json_file)
-waf_policy_file_path = (base_path / "../json/waf-policy.schema").resolve()
-with open(waf_policy_file_path) as json_file:
-    waf_policy_schema = json.load(json_file)
+content_filter_profile_file_path = (base_path / "./json/waf-policy.schema").resolve()
```
Comment (Aviv-Galmidi) [inline curiefense/curieconf/server/curieconf/confserver/v1/api.py:342]: We can also change the name of the scema from `waf-policy` to `content-filter-profile`

File: curiefense/curieconf/server/curieconf/confserver/v1/api.py [hunk 3]
```diff
@@ -334,27 +339,27 @@ def validateJson(json_data, schema_type):
...
 tagrules_file_path = (base_path / "./json/tag-rules.schema").resolve()
 with open(tagrules_file_path) as json_file:
     tagrules_schema = json.load(json_file)
 flowcontrol_file_path = (base_path / "../json/flow-control.schema").resolve()
 with open(flowcontrol_file_path) as json_file:
     flowcontrol_schema = json.load(json_file)
-waf_rule_file_path = (base_path / "../json/waf-rule.schema").resolve()
-with open(waf_rule_file_path) as json_file:
-    waf_rule_schema = json.load(json_file)
+content_filter_rule_file_path = (base_path / "./json/waf-rule.schema").resolve()
```
Comment (Aviv-Galmidi) [inline curiefense/curieconf/server/curieconf/confserver/v1/api.py:351]: We can also change the name of the scema from `waf-rule` to `content-filter-rule`

File: e2e/test_e2e.py [hunk 4]
```diff
@@ -1397,25 +1398,28 @@ def test_nondefault_aclfilter_passthroughall(self, target, securitypolicy_config
...
         )
         assert not target.is_reachable(
-            "/waf-short-headers/", headers={"Long-header": "0123456789" * 5 + "A"}
+            "/content-filter-short-headers/",
+            headers={"Long-header": "0123456789" * 5 + "A"},
         )
 
 
-# --- WAF Policies tests (formerly WAF profiles) ---
+# --- Content Filter Profiles tests (formerly WAF profiles) ---
```
Comment (Aviv-Galmidi) [inline e2e/test_e2e.py:1419]: We can remove the `(formerly WAF profiles)` part, it's not needed

**Review by Freakachoo — APPROVED** [review]

File: curiefense/curieconf/server/curieconf/confserver/v1/api.py (hunk 2 above)
Comment (Ronyk1) [reply curiefense/curieconf/server/curieconf/confserver/v1/api.py]: Changed

File: curiefense/curieconf/server/curieconf/confserver/v1/api.py (hunk 3 above)
Comment (Ronyk1) [reply curiefense/curieconf/server/curieconf/confserver/v1/api.py]: Changed

File: e2e/test_e2e.py (hunk 4 above)
Comment (Ronyk1) [reply e2e/test_e2e.py]: Removed

**Review by flaper87 — APPROVED** [review]

**Pull request merged**